### PR TITLE
Empty observable cases fixed for Task conversion

### PR DESCRIPTION
**Pull request by mlegenhausen**
See #61

### CODE REVIEW

File: test/ReaderObservable.ts [hunk 1]
```diff
@@ -292,14 +299,14 @@ describe('ReaderObservable', () => {
...
+    const x1 = await _.run(pipe(_.of('a'), _.chainIOK(f)), undefined)
+    assert.deepStrictEqual(x1, 1)
   })
 
   it('chainTaskK', async () => {
     const f = (s: string) => R.of(s.length)
-    const x = await _.run(pipe(_.of('a'), _.chainTaskK(f)), undefined)
-    assert.deepStrictEqual(x, 1)
+    const x1 = await _.run(pipe(_.of('a'), _.chainTaskK(f)), undefined)
+    assert.deepStrictEqual(x1, 1)
```
Comment (mlegenhausen) [inline test/ReaderObservable.ts]: Needs to be reverted.

Comment (gcanti) [reply test/ReaderObservable.ts]: What do you mean?

Comment (mlegenhausen) [reply test/ReaderObservable.ts]: Just a note for my future me 😅

File: src/Observable.ts [hunk 2]
```diff
@@ -455,4 +455,26 @@ export const bind = <K extends string, A, B>(
...
+          resolve(result)
+        }
+      }
+    })
+  })
+
+/**
+ * @since 0.6.15
+ */
+export const toTaskOption = <A>(o: Observable<A>): Task<O.Option<A>> => () => o.toPromise().then(O.fromNullable)
```
Comment (gcanti) [inline src/Observable.ts]: I would return `Task<O.Option<NonNullable<A>>>` here

File: src/ReaderObservable.ts [hunk 3]
```diff
@@ -580,9 +580,15 @@ export const bindW: <K extends string, R2, A, B>(
...
 /**
  * @since 0.6.6
  */
-export const toReaderTask = <R, A>(ma: ReaderObservable<R, A>): ReaderTask<R, A> => r => () => run(ma, r)
+export const toReaderTask = <R, A>(ma: ReaderObservable<R, A>): ReaderTask<R, A> => r => T.toTask(ma(r))
+
+/**
+ * @since 0.6.15
+ */
+export const toReaderTaskOption = <R, A>(ma: ReaderObservable<R, A>): ReaderTask<R, O.Option<A>> => r =>
```
Comment (gcanti) [inline src/ReaderObservable.ts:593]: `: ReaderTask<R, O.Option<NonNullable<A>>>`

**Review by gcanti — CHANGES_REQUESTED** [review]

**Comment by gcanti** [comment]
@mlegenhausen could you please update the changelog too?

File: src/Observable.ts (hunk 2 above)
Comment (mlegenhausen) [reply src/Observable.ts]: Good point I think it should be possible to distinguish between an empty `Observable` and an `Observable` with one element that is nullable.

**Review by gcanti — APPROVED** [review]

**Comment by gcanti** [comment]
Thanks @mlegenhausen, released.

**Comment by mlegenhausen** [comment]
Thanks :)

**Pull request merged**